pkg/pagination/pagination.go

```diff
@@ -20,6 +20,31 @@ type Page struct {
 	PageNumber int `json:"page"`
 }
 
+// NewPage creates a new Page with the given pageNumber and pageSize.
+func NewPage(pageNumber int, pageSize int) Page {
+	return Page{
+		PageSize:   pageSize,
+		PageNumber: pageNumber,
+	}
+}
+
+// NewPageFromRef creates a new Page from pointers to pageNumber and pageSize.
+// Useful for handling query parameters.
+func NewPageFromRef(pageNumber *int, pageSize *int) Page {
```

Comment:
IMO if you want this utility you should move it to `pkg/framework/commonhttp`, it's an API specific thing we do

Reply:
I agree, but this is true for `PaginatedResponse` true + `NewPageFromRef` doesn't enforce is a query param, we would need to make an interface for it.